server/pulp/server/webservices/views/events.py

```diff
@@ -0,0 +1,135 @@
+        event_types = params.get('event_types', None)
+
+        manager = factory.event_listener_manager()
+        event = manager.create(notifier_type_id, notifier_config, event_types)
+
+        link = {'_href': reverse('event_resource',
+                kwargs={'event_listener_id': event['id']})}
+        event.update(link)
+
+        response = generate_json_response_with_pulp_encoder(event)
+        redirect_response = generate_redirect_response(response, link['_href'])
+        return redirect_response
+
+
+class EventResourceView(View):
+    """
+    Views for a single event listener.
+    """
+
+    @auth_required(authorization.READ)
+    def get(self, request, event_listener_id):
+        """
+        Retrieve a specific event listener.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param event_listener_id: id for the requested event listener
+        :type event_listener_id: str
+
+        :return: Response containing the event listener
+        :rtype: django.http.HttpResponse
+        """
+        manager = factory.event_listener_manager()
+
+        listener = manager.get(event_listener_id)  # will raise MissingResource
+        link = {'_href': reverse('event_resource',
+                kwargs={'event_listener_id': listener['id']})}
+        listener.update(link)
+        return generate_json_response_with_pulp_encoder(listener)
+
+    @auth_required(authorization.DELETE)
+    def delete(self, request, event_listener_id):
+        """
+        Delete an event listener.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param event_listener_id: id for the requested event listener
+        :type event_listener_id: str
+
+        :return: An mepty response
+        :rtype: django.http.HttpResponse
+        """
+        manager = factory.event_listener_manager()
+
+        manager.delete(event_listener_id)  # will raise MissingResource
+        return generate_json_response(None)
+
+    @auth_required(authorization.UPDATE)
+    def put(self, request, event_listener_id):
+        """
+        Update a specific event listener.
+
+        :param request: WSGI request object
+        :type request: django.core.handlers.wsgi.WSGIRequest
+        :param event_listener_id: id for the requested event listener
+        :type event_listener_id: str
+
+        :return: Response containing the event listener
+        :rtype: django.http.HttpResponse
+        """
+        params = request.body_as_json
+
+        notifier_config = params.get('notifier_config', None)
+        event_types = params.get('event_types', None)
+
+        manager = factory.event_listener_manager()
+        listener = manager.update(event_listener_id, notifier_config=notifier_config,
+                                  event_types=event_types)
+
+        link = {'_href': reverse('event_resource',
```

Comment:
Since this happens 4 different times in the same module, and is a bit complex for 3 lines of code, I'd be inclined to pull it out to its own module-level function. Then if we ever change the view name, change the way we create the link, etc., that change can happen in one place. Something simple like:

```
def _add_link(listener):
  link = {'_href': reverse('event_resource',
          kwargs={'event_listener_id': listener['id']})}
  listener.update(link)
```